### ISSUE TITLE
osu!updater not carrying over startup arguments

### ISSUE THREAD

**Issue by tsunyoku** [ppy/osu-stable-issues#939]
There are various arguments you can provide osu! that change functionality, for example tourney client. In this instance, I'm going to talk specifically about the `devserver` argument. This is particularly problematic as this argument will change the endpoints used to connect to Bancho, you can probably see where I'm going with this.

It's an easily reproducible issue, have your osu!updater run on startup for any reason but make sure your game is using arguments (`devserver` one is most important to test with). Once the updater is done and starts the new process, you'll see that the startup arguments you used originally were not preserved in the new process.

I have a video example here: https://twitter.com/tsunyoku/status/1476055489754636288?s=20&t=2DLIpUTxO2VyJlaWpUCk6A

This is particularly a problem as it can cause accidental multiaccounts with the `devserver` flag, which has happened to me and also a friend of mine. I reported this bug to support nearly a year ago and it still isn't fixed so I figured I should open an issue.

**Comment by peppy** [comment]
Sorry this is out of scope. Be more careful.

**Comment by xxkat** [comment]
Tsunyoku here has made an extensive bug report which not only goes into great detail but also provides a video showing the bug being reproduced, this issue has not only resulted in his account being restricted but also his friends yet, despite all the evidence of such a bug, neither are unrestricted and the best response.

And you can only respond by saying "be more careful"?

**Comment by xxkat** [comment]
The fix is literally slightly modifying a single line to this.

![31d5fa7c-22ab-48df-b5c2-3409ba5473d3](https://user-images.githubusercontent.com/8146687/152700880-e9ccaf8d-b23a-44af-96fd-6f05a5166c5f.png)

**Comment by eren031** [comment]
> Sorry this is out of scope. Be more careful.

I've been restricted for the very same bug, for "multi-accounting" when I wasn't, and when I had asked osu support to clarify, they would never respond.
I'm sure hundreds or thousands of others had been affected by this bug.
Fix it.

**Comment by lenforiee** [comment]
> Sorry this is out of scope. Be more careful.

Really? only thing you can say is "be more careful"??? 
A year ago i was victim of this exact bug as well while i just wanted to create a osu account on some private server, after 4 months i realised what happened but luckily osu staff banned my not intentional "multi-account" but the fact this bug exist **for more than a year** at this point, it has been reported numerous of times and it’s still not fixed is ridiculous, get your self together.

**Comment by RealistikDash** [comment]
> The fix is literally slightly modifying a single line, changing it to this.
> 
> ![31d5fa7c-22ab-48df-b5c2-3409ba5473d3](https://user-images.githubusercontent.com/8146687/152700880-e9ccaf8d-b23a-44af-96fd-6f05a5166c5f.png)

I generally thing directly relaying the flags osu! was orignally started with would be a smart idea. Devserver aside, it would make the whole flags system more convenient for other functions and would avoid weird cases where the game restarts without the flags. Now with a flag that is starting to be used by regular users (who may not be as technologically literate), such weird and unusual behaviour by the client may cause issues that in the end, osu!support would have to assist with.

**Comment by 7ez** [comment]
> Sorry this is out of scope. Be more careful.

i once almost made a multi account on bancho and you're telling me to be careful?? the updater literally runs for like half a millisecond so i can't tell if i launched osu with devserver or not

**Comment by xSkyyy** [comment]
peppy this is an issue for a lot of people that has a really easy fix. please sort your shit.

**Comment by Sensokaku** [comment]
Valid Reports:

Things that affect your daily enjoyment of the game.
Things that don't involve implementing large new features (quick fixes only)
Things that you wonder "why isn't this fixed" but it has never got visibility.

> Things that affect your daily enjoyment of the game.

This is a Valid Report.

**Comment by peppy** [comment]
> > Sorry this is out of scope. Be more careful.
> 
> i once almost made a multi account on bancho and you're telling me to be careful?? the updater literally runs for like half a millisecond so i can't tell if i launched osu with devserver or not

Can you explain this? The updater should only be running if you tell it to. If there is a scenario where it's running without your permission, that is something I'm willing to look at fixing.

**Comment by peppy** [comment]
I'm going to close this thread due to the nature of the people who are replying, but feel free to email me with a scenario where the updater runs on startup without your permission and I'll investigate this further.

**Comment by peppy** [comment]
I've checked on this and the only cases where this can occur is if the game is corrupted or a manual invocation is triggered via the method documented in the OP (ie. it can't happen unless you force it). I'm against changing this pathway if it can be avoided as the intention is to make osu! start at all costs. Passing command line arguments could prohibit this from working (ie. if the user accidentally added arguments which are causing the startup process to fail).

Again, if you have a more specific example please shoot me an email and I'll reconsider.

**Comment by peppy** [comment]
In email correspondence it was understood that this can happen after a crash, specifically. While the rate at which osu! crashes is super low these days, I've fixed that pathway in the next release.
